### PR TITLE
feat: add 202 status code

### PR DESCRIPTION
**Pull request by corneliu-petrescu**
Adds minimal support for Async API responses as per: https://jsonapi.org/recommendations/#asynchronous-processing
 - Standard Type for modeling Job concept
 - 202 response validation with Content-Location header
 - 303 response validation with Location header

### CODE REVIEW

File: docs/standards/rest.md [hunk 1]
```diff
@@ -598,6 +598,11 @@ All status codes must be listed in this section or as a part of the [JSON-API Sp
 
 In addition to POST and DELETE for individual resources, our services may respond to a _collection_ POST or PATCH request with a 204 as a response to bulk resource creation or modification. This is an extension to the JSON API standard.
 
+
+### 202 - Accepted
+
+A request has been accepted for processing, but the processing has not been finished yet. Used for Async Actions. It is recommended to return a Location header as a way of letting the client know where to query for the status. Response Body is not mandatory. Available both for POST, PATCH and DELETE
```
Comment (jgresty) [inline docs/standards/rest.md:604]: I think we should enforce a location header is present for this status code

File: components/common.yaml [hunk 2]
```diff
@@ -222,6 +222,34 @@ schemas:
...
+    properties:
+      status:
+        type: string
+    additionalProperties: false
+    required:
+      - status
+
+  JobResponse:
+    type: object
+    description: async job response
```
Comment (jgresty) [inline components/common.yaml:237]: These descriptions could be better. These will be displayed in our public documentation so need to be more descriptive

**Review by jgresty — APPROVED** [review]
Make sure we follow up with rules enforcing the response body

**Review by jgresty — APPROVED** [review]
Only a couple of minor points, otherwise lgtm

File: components/common.yaml [hunk 3]
```diff
@@ -222,6 +221,44 @@ schemas:
     pattern: "^((([0-9]{4})-([0-1][0-9]))-((3[01])|(0[1-9])|([12][0-9]))(~(wip|work-in-progress|experimental|beta))?)$"
     example: "2021-06-04"
 
+  JobAttributes:
+    type: object
+    description: async job attributes
```
[the remark this answers is not in the document]
Comment (corneliu-petrescu) [reply components/common.yaml:226]: I'll remove the descriptions for now

File: components/common.yaml [hunk 4]
```diff
@@ -222,6 +221,44 @@ schemas:
...
+    properties:
+      status:
+        type: string
```
[the remark this answers is not in the document]
Comment (jgresty) [reply components/common.yaml:228]: This field could still benefit from a description

File: docs/standards/rest.md [hunk 5]
```diff
@@ -598,6 +598,16 @@ All status codes must be listed in this section or as a part of the [JSON-API Sp
...
 
+
+### 202 - Accepted
+
+A request has been accepted for processing, but the processing has not been finished yet. Used for Async Actions. It is mandatory to return a location header as a way of letting the client know where to query for the status. Response Body is not mandatory. Available for POST, PATCH and DELETE.
+The recommendation is to follow the standard as per: [JSON-API#Async-Processing](https://jsonapi.org/recommendations/#asynchronous-processing)
+
+
+### 303 - See Other
+It is mandatory to return a location header as a way of letting the client know where the resource requested is located.
```
Comment (jgresty) [inline docs/standards/rest.md:609]: This doesn't really describe the purpose of this status

**Pull request merged**